src/languages/php.js

```diff
@@ -263,13 +264,80 @@ export default function(hljs) {
     "stdClass"
   ];
 
+  /** Dual-case keywords
+   *
+   * ["then","FILE"] =>
+   *     ["then", "THEN", "FILE", "file"]
+   *
+   * @param {string[]} items */
+  const dualCase = (items) => {
+    /** @type string[] */
+    const result = [];
+    items.forEach(item => {
+      result.push(item);
+      if (item.toLowerCase() === item) {
+        result.push(item.toUpperCase());
+      } else {
+        result.push(item.toLowerCase());
+      }
+    });
+    return result;
+  };
+
   const KEYWORDS = {
     keyword: KWS,
-    literal: LITERALS,
-    built_in: BUILT_INS
+    literal: dualCase(LITERALS),
+    built_in: BUILT_INS,
+  };
+
+  /**
+   * @param {string[]} items */
+  const normalizeKeywords = (items) => {
+    /** @type string[] */
+    const result = [];
+    items.forEach(item => {
+      item.replace(/\|0$/, "");
+      result.push(item);
+    });
+    return result;
```

Comment:
Use `map` for transforms?

Reply:
Also this needs to match `\d+` not just `0`... so it's future proof.

Reply:
I think this is all I see.  I might step in after and make a negative look-ahead list helper... but this looks pretty clean right now.

Reply:
@joshgoebel done

> I think this is all I see. I might step in after and make a negative look-ahead list helper... but this looks pretty clean right now.

Good idea